### PR TITLE
Fix domain slot schema

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- Allow arbitrary keys under a slot in the domain as was allowed in 2.x
- closes https://github.com/RasaHQ/rasa/issues/10765

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/shared/core/test_domain.py [hunk 1]
```diff
@@ -859,6 +859,10 @@ def test_load_domain_with_entity_roles_groups():
     assert "origin" in domain.roles["GPE"]
     assert "destination" in domain.roles["GPE"]
 
+def test_load_domain_with_custom_slot():
+    domain = Domain.load("data/test_domains/custom_slot_domain.yml")
+
+    assert domain.slots is not None
```
Comment (joejuzl) [inline tests/shared/core/test_domain.py:865]: Shall we check that the limit param made it to the slot class too?

File: rasa/shared/utils/schemas/domain.yml [hunk 2]
```diff
@@ -47,6 +47,8 @@ mapping:
       regex;([A-Za-z]+):
         type: "map"
         mapping:
+          regex;([A-Za-z]+):
```
Comment (joejuzl) [inline rasa/shared/utils/schemas/domain.yml:50]: Are we sure that this doesn't short-circuit the other lower one? i.e. do we still enforce that `influence_conversation` is type `bool`?

File: tests/shared/core/test_domain.py (hunk 1 above)
Comment (melindaloubser1) [reply tests/shared/core/test_domain.py:865]: oh yes sorry I never finished this test and forgot 🤦‍♀️

File: rasa/shared/utils/schemas/domain.yml (hunk 2 above)
Comment (melindaloubser1) [reply rasa/shared/utils/schemas/domain.yml:50]: I'll check & also add a test that the validator yells when type is wrong

Comment (melindaloubser1) [reply rasa/shared/utils/schemas/domain.yml:50]: It is still enforced, added tests 👍

File: tests/shared/core/test_domain.py (hunk 1 above)
Comment (melindaloubser1) [reply tests/shared/core/test_domain.py]: updated

**Review by joejuzl — APPROVED** [review]
💯

File: rasa/shared/utils/schemas/domain.yml (hunk 2 above)
Comment (wochinge) [inline rasa/shared/utils/schemas/domain.yml:50]: can't we just do `allowempty: True`? See docs here https://pykwalify.readthedocs.io/en/unstable/validation-rules.html#allowempty

Comment (melindaloubser1) [reply rasa/shared/utils/schemas/domain.yml:50]: Indeed! I only discovered where these docs were after  the fact so I went off of the rest of the schema, but this makes more sense

File: rasa/shared/utils/schemas/domain.yml [hunk 3]
```diff
@@ -47,6 +47,8 @@ mapping:
...
+            type: "any"
```
Comment (melindaloubser1) [inline rasa/shared/utils/schemas/domain.yml:51]:
```suggestion
          allowempty: True
```

Comment (melindaloubser1) [inline rasa/shared/utils/schemas/domain.yml:51]:
```suggestion
        type: "map"
        allowempty: True
        mapping:
```

**Pull request merged**